### PR TITLE
feat: add `ftl build` command

### LINKED ISSUE #597: `ftl build [<dir>]` will load the `ftl.toml` and execute the appropriate code to build a module in a particular language, eg. `mvn compile`

### PR DESCRIPTION
**Pull request by wesbillman**
Fixes #597

This currently only supports building `kotlin` modules. `go` support will be added in the future.

I'm not 100% sure if this is the best way to activate `hermit` as part of the build but we can discuss.

Create new module and build:
```bash
ftl init kotlin services alice

ftl build services/ftl-module-alice
info: Building kotlin module 'alice'
info: Using build command 'source ../bin/activate-hermit && mvn compile'
```

With log-level=debug
```bash
ftl build services/ftl-module-alice --log-level=debug
info: Building kotlin module 'alice'
info: Using build command 'source ../bin/activate-hermit && mvn compile'
debug: ../bin/activate-hermit: line 61: /bin/hermit: No such file or directory
debug: ../bin/activate-hermit: line 20: /bin/hermit: No such file or directory
debug: Hermit environment  activated
debug: [INFO] Scanning for projects...
debug: [INFO]
debug: [INFO] ------------------------< ftl:ftl-module-alice >------------------------
debug: [INFO] Building ftl-module-alice 1.0-SNAPSHOT
debug: [INFO] --------------------------------[ jar ]---------------------------------
debug: [INFO]
debug: [INFO] --- maven-dependency-plugin:3.2.0:copy (default) @ ftl-module-alice ---
debug: [INFO] Configured Artifact: xyz.block:ftl-generator:jar-with-dependencies:1.0-SNAPSHOT:jar
```

Default build command can be overridden in `ftl.toml`

```toml
module = "alice"
language = "kotlin"
build = "source ../bin/activate-hermit && mvn clean compile"
deploy = [
  "target/main",
  "target/classes",
  "target/dependency",
  "target/classpath.txt",
]
```
```bash
ftl build services/ftl-module-alice
info: Building kotlin module 'alice'
info: Using build command 'source ../bin/activate-hermit && mvn clean compile'
```

Example errors:

Building a `go` module:
```bash
ftl build echo
ftl: error: unable to build. unknown language "go"
```

`toml` not found
```bash
ftl build bogus
ftl: error: [<base>]: stat /Users/wesbillman/dev/ftl/examples/bogus: no such file or directory
```

### CODE REVIEW

File: cmd/ftl/cmd_build.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		return errors.Errorf("unable to build. unknown language %q", config.Language)
+	}
+}
+
+func (b *buildCmd) buildKotlin(ctx context.Context, config moduleconfig.ModuleConfig) error {
+	logger := log.FromContext(ctx)
+
+	buildCmd := config.Build
+	if buildCmd == "" {
+		buildCmd = "source ../bin/activate-hermit && mvn compile"
```
Comment (alecthomas) [inline cmd/ftl/cmd_build.go:37]: You'll have to check if the `../bin/hermit` directory exists, as it's optional.

**Review by alecthomas — APPROVED** [review]
Glorious!

Weird that the Hermit activation is throwing an error...

File: cmd/ftl/cmd_build.go (hunk 1 above)
Comment (alecthomas) [reply cmd/ftl/cmd_build.go:37]: Actually I would remove the Hermit activation completely. We should assume that the user has either already activated the environment, or isn't using it.

Comment (wesbillman) [reply cmd/ftl/cmd_build.go:37]: Yeah I was curious about that. In our "default" build would we assume it's there? Or, should I just always check if there is one and if so go ahead and add the activation?

**Pull request merged**